src/fillTank.test.js

```diff
@@ -1,11 +1,104 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared', () => {
+    expect(fillTank).toBeInstanceOf(Function);
+  });
+
+  it('If amount is empty, then full tank is ordered', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+
+    fillTank(customer, 50);
 
+    expect(customer.money).toBe(1400);
+    expect(customer.vehicle.fuelRemains).toBe(40);
   });
 
-  // write tests here
+  it(
+    'If amount is greater than the tank can accommodate,'
+    + ' pour only what will fit.',
+    () => {
+      const customer = {
+        money: 3000,
+        vehicle: {
+          maxTankCapacity: 40,
+          fuelRemains: 8,
+        },
+      };
+
+      fillTank(customer, 50, 50);
+
+      expect(customer.money).toBe(1400);
+      expect(customer.vehicle.fuelRemains).toBe(40);
+    }
+  );
+
+  it('ALWAYS fill in only what the client can pay', () => {
+    const customer = {
+      money: 1000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+
+    fillTank(customer, 50, 30);
```

Comment:
Ensure that the function correctly handles the scenario where the customer can only pay for a portion of the requested fuel amount, and adjusts the fuel poured accordingly.